extensions/wikia/ArticleComments/classes/ArticleComment.class.php

```diff
@@ -197,84 +192,124 @@ static public function newFromId( $id ) {
+		if ( empty( $this->mFirstRevId ) ) {
+			$this->mFirstRevId = $this->getFirstRevID( DB_MASTER );
+		}
 
-			// get revision objects
-			if ( $this->mFirstRevId ) {
-				$this->mFirstRevision = Revision::newFromId( $this->mFirstRevId );
-				if ( !empty( $this->mFirstRevision ) && is_object( $this->mFirstRevision ) && ( $this->mFirstRevision instanceof Revision ) ) { // fix for FB:15198
-					if ( $this->mLastRevId == $this->mFirstRevId ) {
-						// save one db query by just setting them to the same revision object
-						$this->mLastRevision = $this->mFirstRevision;
-					} else {
-						$this->mLastRevision = Revision::newFromId( $this->mLastRevId );
-						if ( empty( $this->mLastRevision ) || !is_object( $this->mLastRevision ) || !( $this->mLastRevision instanceof Revision ) ) {
-							$result = false;
-						}
-					}
-				} else {
-					$result = false;
-				}
-			} else {
-				$result = false;
-			}
+		return !empty( $this->mFirstRevId );
+	}
 
-			// get user that created this comment
-			if ( $this->mFirstRevision ) {
-				$this->mUser = User::newFromId( $this->mFirstRevision->getUser() );
-				$this->mUser->setName( $this->mFirstRevision->getUserText() );
-			} else {
-				$result = false;
-			}
+	private function loadLastRevId( $useMaster = false ) {
+		if ( empty( $this->mTitle ) ) {
+			return false;
+		}
 
-			if ( empty( $this->mFirstRevision ) || empty( $this->mLastRevision ) ) {
-				wfProfileOut( __METHOD__ );
-				return false;
-			}
+		// Exit early if we've already loaded this
+		if ( !empty( $this->mLastRevId ) ) {
+			return true;
+		}
 
-			$rawText = $this->mLastRevision->getText();
-			$this->parseText( $rawText );
-		} else { // null title
-			$result = false;
+		if ( !$useMaster ) {
+			$this->mLastRevId = $this->mTitle->getLatestRevID();
 		}
 
-		wfProfileOut( __METHOD__ );
-		return $result;
+		// Fall back to master if not on slave or if we wanted master in the first place
+		if ( empty( $this->mLastRevId ) ) {
+			$this->mLastRevId = $this->mTitle->getLatestRevID( Title::GAID_FOR_UPDATE );
+		}
+
+		// Use first rev ID as last resort
+		if ( empty( $this->mLastRevId ) ) {
+			$this->loadFirstRevId();
+			$this->mLastRevId = $this->mFirstRevId;
+		}
+
+		return !empty( $this->mLastRevId );
+	}
+
+	private function loadFirstRevision() {
+		if ( !empty( $this->mFirstRevision ) ) {
+			return true;
+		}
+
+		// get revision objects
+		$this->mFirstRevision = Revision::newFromId( $this->mFirstRevId );
+
+		return !empty( $this->mFirstRevision ) && $this->mFirstRevision instanceof Revision;
+	}
+
+	private function loadLastRevision() {
+		if ( !empty( $this->mLastRevision ) ) {
+			return true;
+		}
+
+		if ( $this->mLastRevId == $this->mFirstRevId ) {
+			// save one db query by just setting them to the same revision object
+			$this->mLastRevision = $this->mFirstRevision;
+		} else {
+			$this->mLastRevision = Revision::newFromId( $this->mLastRevId );
```

Comment:
Consider replacing all of reads of `$this->mLastRevId` and `$this->mFirstRevId` with their corresponding getters so that dependent method calls do not have to get executed in a specific order in order for everything to work.